src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -195,7 +195,7 @@ private void WritePrimitiveValue(TypeDesc typeDesc, SourceInfo source, out Type
         }
 
         [RequiresUnreferencedCode("Calls WriteCheckDefault")]
-        private void WritePrimitive(string method, string name, string? ns, object? defaultValue, SourceInfo source, TypeMapping mapping, bool writeXsiType, bool isElement, bool isNullable)
```

Comment:
**writeXsiType** was always false in all call sites.
**isElement** was not used also